core/src/main/java/org/aerogear/mobile/core/metrics/publisher/NetworkMetricsPublisher.java

```diff
@@ -22,23 +22,23 @@
     private HttpRequest httpRequest;
```

Comment:
make the members final too

Reply:
I actually once complained about the class members that are private/static/final. We're designing an SDK, which could be extended by the users and I think leaving room for extending things could be nice. This is a wider discussion though. ...and we can always convert stuff to protected and non final later.
Any devex guys around here?

Reply:
regarding the members:
* if members are only instantiated by the ctor, why on earth should they be mutable ? this quickly leads to code that easier enforces bugs.   since they are private inheritance also does not make use of them ... so: make them final.  In a decent java dialect, like Scala, this would be `val`, not `var` 

I think you are more asking: 
* do we want users to extend from these concrete classes?

looking at the code I'd doubt it, but having some "abstract" base classes (publishers, metrics impls) would easily allow this, while still keeping the code clean